### ISSUE TITLE
[Doc] Update all Ray Train Examples that needs Metrics Aggregation

### ISSUE THREAD

**Issue by woshiyyya** [ray-project/ray#38760]
### Description

In PR #38694, we changed the dataset splitting default behavior from splitting "train" only to split all datasets.

Therefore, we must ensure that all Ray Train examples either
- aggregate metrics from workers with utilities like torchmetrics, accelarate
- or specify not to split the validation set in `DataConfig`.

**Comment by woshiyyya** [comment]
I looked into the examples in [Ray Train Examples](https://docs.ray.io/en/master/train/examples.html). Seems they are either already handled aggregation with 3rd party library (lightning, transformers, accelerate), or does not have evaluation sets.
